src/solvers/z3/CMakeLists.txt

```diff
@@ -2,93 +2,53 @@
 #  * User specified directories
 #  * Environmental var paths
 #  * System default
-function(findz3 versionstr libz3 incz3)
-    find_library(LIBZ3 z3 PATHS "${ARGN}" PATH_SUFFIXES lib bin NO_DEFAULT_PATH)
-    find_path(INCZ3 z3.h PATHS "${ARGN}/include" NO_DEFAULT_PATH)
+if(DEFINED Z3_DIR)
+    set(ENABLE_Z3 ON)
+endif()
 
-    if ((LIBZ3 STREQUAL "LIBZ3-NOTFOUND") OR (INCZ3 STREQUAL "INCZ3-NOTFOUND"))
-        return()
-    endif ()    
-    get_filename_component(Z3_LIBDIR "${LIBZ3}" DIRECTORY)
-    try_run(Z3RUNS Z3COMPILES ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/try_z3.c
-            CMAKE_FLAGS -DINCLUDE_DIRECTORIES=${ARGN}/include
-            LINK_LIBRARIES ${LIBZ3} stdc++ m -pthread
-            
-            COMPILE_OUTPUT_VARIABLE Z3COMPILESTR
-            RUN_OUTPUT_VARIABLE Z3VERSIONSTR)
-
-    if ((NOT Z3RUNS EQ 0) OR (NOT Z3COMPILES))
-        message(FATAL_ERRROR "Failed Z3 compilation compile: ${Z3COMPILESTR}")
-        return()
-    endif ()
+if(EXISTS $ENV{HOME}/z3)
+    set(ENABLE_Z3 ON)
+endif()
 
-    set(${libz3} "${LIBZ3}" PARENT_SCOPE)
-    set(${incz3} "${ARGN}/include" PARENT_SCOPE)
-    set(${versionstr} "${Z3VERSIONSTR}" PARENT_SCOPE)
-endfunction()
 
-if (DEFINED DISABLE_Z3)
-    if (DEFINED ENABLE_Z3)
-        message(SEND_ERROR "Can't request both disable and enable z3")
-    endif ()
-    set(YESWEUSED_DIABLE_Z3, "${DISABLE_Z3}")
-    return()
-endif ()
-
-set(LIBZ3 "LIBZ3-NOTFOUND")
-set(INCZ3 "LIBZ3-NOTFOUND")
+if(ENABLE_Z3)
+    find_library(Z3_LIB z3 PATHS ${Z3_DIR} PATH_SUFFIXES lib bin)
+    find_path(Z3_INCLUDE_DIRS z3.h PATHS ${Z3_DIR} PATH_SUFFIXES include)
```

Comment:
shouldn't there be a `$ENV{HOME}/z3` like the others?

Reply:
Yes, it should.